engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/views/repository-info.html

```diff
@@ -49,9 +49,17 @@ <h4 translate>Basic Information</h4>
       </dd>
 
       <dt translate>Upstream Password</dt>
-      <dd bst-edit-text="repository.upstream_password"
+      <dd bst-edit-custom="repository.upstream_password_exists"
+          readonly="denied('edit_products', product)"
           on-save="save(repository)"
-          readonly="product.redhat || denied('edit_products', product)">
+          formatter="upstreamPasswordFilter"
+          formatter-options="repository"
+          deletable="repository.upstream_password_exists"
+          on-delete="clearUpstreamPassword()"
+          >
+          <input id="upstream_password"
+            name="upstream_password"
```

Comment:
Shouldn't this input be of type password?  I can view the password as I type it.